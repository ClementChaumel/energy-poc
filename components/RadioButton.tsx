import * as React from 'react';
import styled from 'styled-components';
import theme from '@totallymoney/ui/theme';

const RButton = styled.button`
  background-color: white;
  border: solid 1px ${theme.indigo120};
  border-radius: 5px;
  padding: ${theme.spacingS};
  font-size: 1.6rem;
  cursor: pointer;
  text-align: left;
  display: flex;
  align-items: center;

  &:before {
    content: '';
    display: block;
    height: 15px;
    width: 15px;
    background: white;
    border: 1px solid ${theme.indigo120};
    border-radius: 50%;
    margin-right: ${theme.spacingS};
  }

  &:focus:before {
    background-color: ${theme.indigo120};
    box-shadow: inset 0px 0px 0 2px white;
  }
`;
interface RadioButtonTypes {
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  label: string;
  id: string;
}

const RadioButton: React.SFC<RadioButtonTypes> = props => {
  const { onClick, label, id } = props;
  return (
    <RButton type="button" onClick={onClick} id={id}>
      {label}
    </RButton>
  );
};

export default RadioButton;
