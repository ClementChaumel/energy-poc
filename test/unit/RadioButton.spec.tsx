import * as React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import RadioButton from '../../components/RadioButton';

describe('<RadioButton />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<RadioButton id="both" label="Both gas and electricity" onClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
