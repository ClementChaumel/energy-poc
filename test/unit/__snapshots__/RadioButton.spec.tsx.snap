// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders correctly 1`] = `
.c0 {
  background-color: white;
  border: solid 1px #08051d;
  border-radius: 5px;
  padding: 16px;
  font-size: 1.6rem;
  cursor: pointer;
  text-align: left;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:before {
  content: '';
  display: block;
  height: 15px;
  width: 15px;
  background: white;
  border: 1px solid #08051d;
  border-radius: 50%;
  margin-right: 16px;
}

.c0:focus:before {
  background-color: #08051d;
  box-shadow: inset 0px 0px 0 2px white;
}

<button
  className="c0"
  id="both"
  onClick={[Function]}
  type="button"
>
  Both gas and electricity
</button>
`;
